### PR TITLE
Fixes #30730 - Implement systemd notify support for dynflow-sidekiq

### PR DESCRIPTION
**Pull request by adamruzicka**

### CODE REVIEW

File: extras/systemd/dynflow-sidekiq@.service [hunk 1]
```diff
@@ -4,9 +4,9 @@ Documentation=https://theforeman.org
 After=network.target remote-fs.target nss-lookup.target
 
 [Service]
-Type=simple
+Type=notify
 User=foreman
-TimeoutSec=90
+TimeoutSec=300
```
Comment (adamruzicka) [inline extras/systemd/dynflow-sidekiq@.service:9]: On my machine (4cpus, 16gb of ram) the service failed to start in time. Foreman service uses 300 so I set it here as well.

**Review by ekohl — COMMENTED** [review]
Overall :+1: for the general design.

I am wondering if there's a useful status message we can share. Is it useful to share the queues it's handling? Whether it's remote? Most services don't imlpement this, but Apache shows `Total requests: 0; Current requests/sec: 0; Current traffic:   0 B/sec` when you run `systemctl status apache`.

Could you also submit the RPM packaging PR?

**Comment by adamruzicka** [comment]
> I am wondering if there's a useful status message we can share.

Having it display the throughput/stats would be nice, however I'm afraid we'd have to implement this change in Dynflow or somehow hook onto sidekiq's internals. Let me think on this over the weekend if I can come up with something that would be useful

> Is it useful to share the queues it's handling?

Could be to some users. However we'd have to do this sensibly since there is no limitation placed on the number of queues being handled by a single process.

> Whether it's remote? 

What do you mean by this?

> Could you also submit the RPM packaging PR?

On my todo list.

**Comment by ekohl** [comment]
> Could be to some users. However we'd have to do this sensibly since there is no limitation placed on the number of queues being handled by a single process.

Perhaps if it's more than x (3?) then you can print `Handling x queues` instead of the actual values.

> What do you mean by this?

Essentially the value of `::Rails.application.dynflow.config.remote`. Not sure if it's interesting. Perhaps this is always true in production setups and thus irrelevant.

**Comment by adamruzicka** [comment]
> Essentially the value of ::Rails.application.dynflow.config.remote. Not sure if it's interesting. Perhaps this is always true in production setups and thus irrelevant.

In production setups the orchestrator should have it false and workers true. Not sure if that's interesting to anyone

**Comment by ekohl** [comment]
Probably not. And perhaps all of these things should rather be logged during start up and that's sufficient. I certainly don't want to force you to implement the status. Just make you think about the use case.

**Review by ekohl — APPROVED** [review]
:+1: code wise pending packaging merge. My only concern is that we should somehow note that sidekiq 6 has native support for systemd. If we upgrade, we should be sure to use that. Thoughts on how to do so?

**Comment by adamruzicka** [comment]
Sidekiq seems to implement `ready` and `stopping` state notifications and watchdog. Once we start using Sidekiq 6, I will have to look what is the ordering between our notifications and the ones in Sidekiq. For `ready`, we should drop ours since Sidekiq's will happen later, however for `stopping` I'm not sure, but here we should pick the one that happens earlier.

**Comment by ekohl** [comment]
I just noticed you created a new issue rather than reusing mine. Left a note @ https://projects.theforeman.org/issues/30275#note-4 and related both issues. I think that should be sufficient.

**Comment by ehelms** [comment]
Packaging of the gem has been merged. A follow up PR will be needed to update Foreman requirements on the gem and then we can orchestrate merging this.

**Comment by ekohl** [comment]
Merging. Automated tools work better after it's merged.

**Comment by ekohl** [comment]
https://github.com/theforeman/foreman-packaging/pull/5735

**Pull request merged**